source/backend/api/Services/AcquisitionFileService.cs

```diff
@@ -140,8 +140,11 @@ public PimsAcquisitionFile UpdateProperties(PimsAcquisitionFile acquisitionFile)
                 _acquisitionFilePropertyRepository.Delete(deletedProperty);
                 if (deletedProperty.Property.IsPropertyOfInterest.HasValue && deletedProperty.Property.IsPropertyOfInterest.Value)
                 {
-                    int propertyCount = _acquisitionFilePropertyRepository.GetAcquisitionFilePropertyRelatedCount(deletedProperty.PropertyId);
-                    if (propertyCount == 1 && deletedProperty?.Property?.IsPropertyOfInterest == true)
+                    PimsProperty propertyWithAssociations = _propertyRepository.GetAssociations(deletedProperty.PropertyId);
+                    var leaseAssociationCount = propertyWithAssociations.PimsPropertyLeases.Count;
+                    var researchAssociationCount = propertyWithAssociations.PimsPropertyResearchFiles.Count;
+                    var acquisitionAssociationCount = propertyWithAssociations.PimsPropertyAcquisitionFiles.Count;
```

Comment:
maybe we need a method on the property repository that returns a total count of each association.